freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UsersyncPoller.java

```diff
@@ -0,0 +1,110 @@
+package com.sequenceiq.freeipa.service.freeipa.user;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.sequenceiq.cloudbreak.auth.ThreadBasedUserCrnProvider;
+import com.sequenceiq.cloudbreak.auth.altus.Crn;
+import com.sequenceiq.cloudbreak.auth.security.InternalCrnBuilder;
+import com.sequenceiq.cloudbreak.logger.MDCBuilder;
+import com.sequenceiq.freeipa.api.v1.freeipa.user.model.SyncOperationStatus;
+import com.sequenceiq.freeipa.entity.Stack;
+import com.sequenceiq.freeipa.entity.UserSyncStatus;
+import com.sequenceiq.freeipa.service.freeipa.user.model.UmsEventGenerationIds;
+import com.sequenceiq.freeipa.service.stack.StackService;
+
+@Service
+public class UsersyncPoller {
+    @VisibleForTesting
+    static final String INTERNAL_ACTOR_CRN = new InternalCrnBuilder(Crn.Service.IAM).getInternalCrnForServiceAsString();
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UsersyncPoller.class);
+
+    @Inject
+    private ThreadBasedUserCrnProvider threadBasedUserCrnProvider;
+
+    @Inject
+    private StackService stackService;
+
+    @Inject
+    private UserSyncStatusService userSyncStatusService;
+
+    @Inject
+    private UserService userService;
+
+    @Inject
+    private UmsEventGenerationIdsUtil umsEventGenerationIdsUtil;
+
+    @Value("${freeipa.syncoperation.poller.enabled:true}")
+    private boolean enabled;
+
+    @Scheduled(fixedDelayString = "${freeipa.syncoperation.poller.fixed-delay-millis:60000}",
+            initialDelayString = "${freeipa.syncoperation.poller.initial-delay-millis:300000}")
+    public void pollUms() {
+        try {
+            if (enabled) {
+                syncFreeIpaStacks();
```

Comment:
So how many instances of this service are running at the same time? 3? Every one of them is running this loop? or are you choose one instance to run the sync? Should you coordinate between them?

Reply:
I don't know of any way for the FMS services to communicate/coordinate amongst themselves other than through the db. I'm relying on the db to reject simultaneous syncs.